chamber/importers/__init__.py

```diff
@@ -0,0 +1,204 @@
+    delimiter = ';'
+    encoding = 'utf-8'
+
+    def __call__(self, csv_path):
+        """csv_path is a required parameter as calling the function without CSV does not make sense"""
+        self.import_csv(custom_csv_path=csv_path)
+
+    def import_csv(self, custom_csv_path=None):
+        with io.open(self.get_filename(custom_csv_path=custom_csv_path), encoding=self.get_encoding()) as f:
+            reader = csv.reader(f, delimiter=self.get_delimiter())
+            if self.get_skip_header():
+                next(reader, None)
+            self.import_rows(
+                reader,
+                row_count=simple_count(self.get_filename(custom_csv_path=custom_csv_path), encoding=self.get_encoding())
+            )
+
+    def import_rows(self, reader, row_count=None):
+        raise NotImplementedError
+
+    @property
+    def out_stream(self):
+        """
+        By default, output stream is essentially turned off by supplying dummy StringIO.
+        Override this property if you want to direct output somewhere, e.g. in Django commands.
+        """
+        return DummyOutputStream()
+
+    def get_filename(self, custom_csv_path=None):
+        return os.path.join(settings.PROJECT_DIR, custom_csv_path or self.get_csv_path())
+
+    def get_csv_path(self):
+        """Override this in case you need to set the CSV path dynamically."""
+        return self.csv_path
+
+    def get_encoding(self):
+        return self.encoding
+
+    def get_delimiter(self):
+        return str(self.delimiter)
+
+    def get_skip_header(self):
+        return self.skip_header
+
+    def get_fields(self):
+        return tuple(f if isinstance(f, six.text_type) else six.u(f) for f in self.fields)
+
+    def get_fields_dict(self, row):
+        """
+        Returns a dict of field name and cleaned value pairs to initialize the model.
+        Beware, it aligns the lists of fields and row values with Nones to allow for adding fields not found in the CSV.
+        Whitespace around the value of the cell is stripped.
+        """
+        return {k: getattr(self, 'clean_{}'.format(k), lambda x: x)(v.strip() if isinstance(v, six.string_types)
+                                                                    else None)
+                for k, v in zip_longest(self.get_fields(), row)}
+
+
+class BulkCSVImporter(AbstractCSVImporter):
+
+    delete_existing_objects = False  # Set to true if you want to delete all records in the model table
+    batch_size = 100000
+
+    def create_batch(self, chunk):
+        return len(self.model_class.objects.bulk_create(chunk))
+
+    def import_rows(self, reader, row_count=None):
+
+        if self.get_delete_existing_objects():
+            self.model_class.objects.all().delete()
+
+        if self.get_skip_header():
+            row_count -= 1
+
+        batch, created = [], 0
+
+        bar = pyprind.ProgBar(row_count, stream=ProgressBarStream(self.out_stream))
```

Comment:
What will happen, when row_count is None and it will enter ProgBar? Will it not raise some error?